Sources/SpeziOnboarding/OnboardingIdentifiableViewModifier.swift

```diff
@@ -0,0 +1,60 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SwiftUI
+
+/// Wrap `Content` `View` in an `Identifiable` `View`.
+private struct OnboardingIdentifiableView<Content, ID>: View, Identifiable where Content: View, ID: Hashable {
+    /// Unique identifier of the wrapped `View`.
+    let id: ID
+    /// Wrapped `View`.
+    let body: Content
+}
+
+private struct OnboardingIdentifiableViewModifier<ID>: ViewModifier, Identifiable where ID: Hashable {
+    let id: ID
+
+    func body(content: Content) -> some View {
+        OnboardingIdentifiableView(
+            id: self.id,
+            body: content
+        )
+    }
+}
+
+extension View {
+    /// `ViewModifier` assigning an identifier to the `View` it is applied to.
+    /// When applying this modifier repeatedly, the outermost ``onboardingIdentifier(_:)`` counts.
```

Comment:
DocC uses the first line of a documentation as a brief summary and embeds that e.g. into the topics section at the bottom a documentation page to provide some details of a reference type. Any text following the first empty line is put into the overview section of a symbol (see [Writing symbol documentation in your source files](https://developer.apple.com/documentation/xcode/writing-symbol-documentation-in-your-source-files#Add-a-basic-description-for-each-symbol)).

I would suggest to put a single, short and descriptive sentence into the first line of a type or symbol.
Feel free to adapt to your liking.

```suggestion
    /// Assign a unique identifier to a `View` appearing in an `OnboardingStack`.
    ///
    ///  A `ViewModifier` assigning an identifier to the `View` it is applied to.
    /// When applying this modifier repeatedly, the outermost ``onboardingIdentifier(_:)`` counts.
```